core/src/main/java/org/apache/iceberg/metrics/MultiDimensionCounterResult.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.metrics;
+
+import java.util.Map;
+import org.apache.iceberg.metrics.MetricsContext.Unit;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public abstract class MultiDimensionCounterResult {
+  public abstract Map<String, Long> counterResults();
```

Comment:
I would have expected this to be a `Map<String, CounterResult>` since a nested counter is still a valid counter. Also aren't we losing the unit of the counter that way?

Reply:
My fist approach had a Map<String, CounterResult> but then I figured that the Unit is going to be the same for the underlying counters so it's enough to have a Long instead of a CounterResult as the Unit would be redundant.